integration/src/wallets/metamask.ts

```diff
@@ -10,8 +12,9 @@ export function createInfo(): core.HDWalletInfo {
 }
 
 export async function createWallet(): Promise<core.HDWallet> {
-  const wallet = new metamask.MetaMaskHDWallet();
-  await wallet.initialize();
+  const keyring = new core.Keyring();
+  const adapter = metamask.MetaMaskAdapter.useKeyring(keyring);
+  const wallet = await adapter.pairDevice()
```

Comment:
Creating the wallet via the adapter (which is the public interface) now causes the provider to be pulled in from the `@metamask/detect-provider` mock.